src/pytest_allure_spec_coverage/models/collector.py

```diff
@@ -0,0 +1,51 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Abstract collector"""
+import os
+from abc import ABC, abstractmethod
+from typing import Optional, List
+
+import toml
+
+from pytest_allure_spec_coverage.models.scenario import Scenario
+
+
+class Collector(ABC):
+    """
+    Abstract collector
+    """
+
+    config: Optional[dict] = None
+    path_to_config_file = "pyproject.toml"
+
+    def __init__(self):
+        self.read_config()
+
+    @abstractmethod
+    def collect(self) -> List[Scenario]:
+        """
+        Main method of collect scenarios
+        """
+
+    @abstractmethod
+    def validate_config(self):
+        """
+        Collector by default read configuration without validation
+        Each collector needs its own config params
+        """
```

Comment:
```suggestion
        """
        Each collector can have an individual config structure.
        This method will be called at the final stage of the config load to validate its structure
        """
```

Reply:
Also need to add a note about expected behavior if config validation fails. What type of error should be raised and so on.

Reply:
I suggest adding to the docstring the fact that not only validation is taking place here, but also attributes assignment.
Because this method has `validate` in its name, it's not so obvious that it does more.
Or maybe we can split validation and assignment, but it looks like overengineering at the start.

Reply:
maybe `setup_config` looks better?

Reply:
renamed